docs/understand/cmake_packages.rst

```diff
@@ -171,4 +220,156 @@ ROCm CMake Packages
 | MIGraphX  | migraphx | ``migraphx::migraphx``, ``migraphx::migraphx_c``,      |
 |           |          | ``migraphx::migraphx_cpu``, ``migraphx::migraphx_gpu``,|
 |           |          | ``migraphx::migraphx_onnx``, ``migraphx::migraphx_tf`` |
-+-----------+----------+--------------------------------------------------------+
\ No newline at end of file
++-----------+----------+--------------------------------------------------------+
+
+Using CMake Presets
+===================
+
+CMake command-lines depending on how specific users like to be when compiling
+code can grow to unwieldy lengths. This is the primary reason why projects tend
+to bake script snippets into their build definitions controlling compiler
+warning levels, changing CMake defaults (``CMAKE_BUILD_TYPE`` or
+``BUILD_SHARED_LIBS`` just to name a few) and all sorts anti-patterns, all in
+the name of convenience.
+
+Load on the command-line interface (CLI) starts immediately by selecting a
```

Comment:
This is unclear: I think we're saying that the `toolchain` file (and various variables, such as `CMAKE_CXX_COMPILER`) are set _before_ the beginning of the user's `CMake` file?

Reply:
What this is saying, is that supplying a complete toolchain on the CLI requires setting 2-10 variables. To ease providing all these variables, CMake reads a few env vars, and also allows reading an entire script, the toolchain file. But the toolchain is just the tip of the iceberg, because there are many other variables the user may want to set.